src/Compiler/Facilities/AsyncMemoize.fs

```diff
@@ -113,14 +112,24 @@ type internal AsyncLock() =
 
     member _.Semaphore = semaphore
 
-    member _.Do(f) =
```

Comment:
I like the cleanup, but does it still behave the same? Can we really remove the `Task.Run`s and the linked CT source?

Reply:
OMG I see I just lost somehow lost the linked source when reshuffling the code. It was definitely not intentional to remove it.

Please don't assume I know what I'm doing here :D
Funny thing it still was passing for me locally. And I actually build the vsix with it and I'm running VS with it.

As for Task.Run I guess the equivalent would be backgroundTask instead of task CE?

Reply:
There might also be difference in when the cancellation request is checked between the Task.Run and Async.StartAsTask.

Reply:
Tasks seem to behaving quite differently on Linux and Mac. And the tests are unfortunately not good enough to cover everything. 

I can't really say that this is better or worse than the previous version, all I can say is that I've been using the previous one for a while and it mostly worked - although still crashes VS sometimes.

There's definitely a lot of room for improvement and testing.